src/lib_ccx/ccx_encoders_common.c

```diff
@@ -449,15 +448,32 @@ static int write_subtitle_file_header(struct encoder_ctx *ctx, struct ccx_s_writ
 			ret = write_bom(ctx, out);
 			if (ret < 0)
 				return -1;
-			REQUEST_BUFFER_CAPACITY(ctx, strlen (webvtt_header)*3);
-			used = encode_line (ctx, ctx->buffer,(unsigned char *) webvtt_header);
-			ret = write (out->fh, ctx->buffer,used);
-			if(ret < used)
+			for(int i = 0; i < sizeof(webvtt_header)/sizeof(webvtt_header[0]);i++)
 			{
-				mprint("WARNING: Unable to write complete Buffer \n");
-				return -1;
+				header_size += strlen(webvtt_header[i]);
+				if(ccx_options.enc_cfg.line_terminator_lf == 1 && strcmp(webvtt_header[i],"\r\n")==0)
+				{
+					header_size--;
```

Comment:
What's the goal here? If it's just adjust the requested buffer, it's unnecessary complex, we don't care that the buffer has a few extra bytes allocated.